src/main/java/io/reactivex/internal/operators/parallel/ParallelJoin.java

```diff
@@ -238,9 +238,7 @@ void drainLoop() {
 
                     boolean empty = true;
 
-                    for (int i = 0; i < n; i++) {
-                        JoinInnerSubscriber<T> inner = s[i];
-
+                    for (JoinInnerSubscriber<T> inner : s) {
```

Comment:
You might want to leave it as it was before since that way you won't allocate an iterator